pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -389,11 +389,10 @@ func (oe *operationExecutor) DetachVolume(
 }
 
 func (oe *operationExecutor) VerifyVolumesAreAttached(
-	attachedVolumes []AttachedVolume,
-	nodeName types.NodeName,
+	attachedVolumesByNode map[types.NodeName][]AttachedVolume,
 	actualStateOfWorld ActualStateOfWorldAttacherUpdater) error {
 	volumesAreAttachedFunc, err :=
-		oe.operationGenerator.GenerateVolumesAreAttachedFunc(attachedVolumes, nodeName, actualStateOfWorld)
+		oe.operationGenerator.GenerateVolumesAreAttachedFunc(attachedVolumesByNode, actualStateOfWorld)
 	if err != nil {
```

Comment:
we are queuing up this task with blank `volumeName` which will cause this task to run parallely with previous execution of same task, if previous task hasn't finished. Coupled with default `5s` polling - I think there is a real chance 2 same tasks will be running parallely.

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_executor.go#L401